genesis/process/metaGenesisBlockCreator.go

```diff
@@ -0,0 +1,322 @@
+package process
+
+import (
+	"bytes"
+	"encoding/hex"
+	"math"
+	"math/big"
+	"sort"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/genesis/process/disabled"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/preprocess"
+	"github.com/ElrondNetwork/elrond-go/process/coordinator"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/process/factory/metachain"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract/builtInFunctions"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract/hooks"
+	processTransaction "github.com/ElrondNetwork/elrond-go/process/transaction"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmFactory "github.com/ElrondNetwork/elrond-go/vm/factory"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// CreateMetaGenesisBlock will create a metachain genesis block
+func CreateMetaGenesisBlock(arg ArgsGenesisBlockCreator) (data.HeaderHandler, error) {
```

Comment:
can you make a component a struct with a New Function and a Create function ? which does this thing ?

Reply:
Done that in the first thing but decided to do this pattern instead as it will solely be used when testing GenesisBlockCreator structure (that have the pointers) With the structures the code becomes larger and not necessarily more testable (as I need to call the constructor the new argument just before the Create method call) So it is better to use a factory method here against the abstract factory pattern.

Reply:
ok. can stay.